engine/hid/src/dmsdk/hid/hid.h

```diff
@@ -0,0 +1,589 @@
+// Copyright 2020 The Defold Foundation
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#ifndef DMSDK_HID_H
+#define DMSDK_HID_H
+
+#include <stdint.h>
+#include <dmsdk/hid/hid_native_defines.h>
+
+/*# SDK Hid API documentation
+ * Used to add input to the engine
+ *
+ * @document
+ * @name Hid
+ * @namespace dmHid
+ * @path engine/hid/src/dmsdk/hid/hid.h
+ */
+
+namespace dmHID
+{
+    /*# HID context handle
+     * @typedef
+     * @name dmHID::HContext
+     */
+    typedef struct Context* HContext;
```

Comment:
We don't have a "typedef" right now, but we'll have to add it.
Same goe for "constant"